lms/static/scripts/frontend_apps/components/ErrorDialogApp.js

```diff
@@ -0,0 +1,65 @@
+import { Fragment, createElement } from 'preact';
+import { useContext } from 'preact/hooks';
+
+import { Config } from '../config';
+
+import ErrorDisplay from './ErrorDisplay';
+import Dialog from './Dialog';
+
+/**
+ * Error dialog displayed when authorization with a third-party API via OAuth
+ * fails.
+ *
+ */
+export default function ErrorDialogApp() {
+  const {
+    ErrorDialog: { errorCode, errorDetails = '' },
+  } = useContext(Config);
+
+  const error = { code: errorCode, details: errorDetails };
+
+  const title = (() => {
```

Comment:
For initializing a set of related variables based on an enum value we generally prefer a `switch`, like so:

```js
  let title;
  let message;

  switch (errorCode) {
    case 'reused_tool_guid':
      title =
        'The Hypothesis credentials your install is using may have already been used in a different LMS instance';
      message = 'Reused tool_consumer_instance_guid';
      break;
    default:
      title = 'An error occurred';
      message = 'Unknown error occurred';
  }
```

There are pros and cons to this, but it has the benefit of grouping all the initializations for the same enum value. It is also just the most obvious language construct to use for this purpose.

I probably should have recommended this for the recent changes in `OAuth2RedirectErrorApp` as well.